src/Stratis.Bitcoin.Features.Consensus/CoinViews/CoinViewStack.cs

```diff
@@ -39,10 +42,14 @@ public IEnumerable<CoinView> GetElements()
             CoinView current = this.Top;
             while (current is IBackedCoinView)
             {
-                yield return current;
+                if (current != null)
```

Comment:
This failed in one of the tests?

Reply:
I got a null value at the end of the list.

Reply:
Edit: I got a null value in the ienumerable from the end of the coinviews stack not having an inner. I will check if we truly need the null check inside the while loop. I think that one can be removed but keep the one before the end.